watertap/examples/flowsheets/RO_with_energy_recovery/RO_with_energy_recovery.py

```diff
@@ -257,7 +257,8 @@ def solve(blk, solver=None, tee=False):
     if solver is None:
         solver = get_solver()
     results = solver.solve(blk, tee=tee)
-    assert_optimal_termination(results)
+
+    return results
```

Comment:
Any reason not do just use the `_default_optimize` function and leave this file as-it-is?

Reply:
Not sure about this one. `_default_optimize` existed in `parameter_sweep.py` before I started making changes. @eyoung55 what are your thoughts?

Reply:
My bad, I misunderstood your comment previously. The current test suite does pass when using `_default_optimize` in `parameter_sweep.py`. So it can be left in its original state. I know you added the option to check for optional termination condition in https://github.com/watertap-org/watertap/pull/342#issue-1147740717. I am fine either way.